Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp

```diff
@@ -486,6 +497,38 @@ static void webkit_website_data_manager_class_init(WebKitWebsiteDataManagerClass
             nullptr, nullptr,
             FALSE,
             static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY)));
+
+    /**
+     * WebKitWebsiteDataManager:per-origin-storage-quota:
+     *
+     * Max size in bytes of origin data storage.
+     * Value of 0 means no change so whatever the default value is it will be used.
+     *
+     * Since: 2.42
+     */
+    g_object_class_install_property(
+        gObjectClass,
+        PROP_PER_ORIGIN_STORAGE_QUOTA,
+        g_param_spec_uint64("per-origin-storage-quota",
+            nullptr, nullptr,
+            0, G_MAXUINT64, 0,
+            static_cast<GParamFlags>(WEBKIT_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY)));
+
+    /**
+     * WebKitWebsiteDataManager:origin-storage-ratio:
+     *
+     * The ratio(0-1) which is used to calculate the origin data storage base on disk space.
```

Comment:
```suggestion
     * The ratio (0-1) which is used to calculate the origin data storage base on disk space.
```

We need a little more detail in this comment. 0.1 would mean the database is allowed to consume 10% of the user's disk space, correct? So values should be very small, e.g. 0.02 would be a reasonable default.

What happens if you do not change the default value and leave it at 0? This should be documented too. I doubt it means "no limit," does it?

Reply:
> What happens if you do not change the default value and leave it at 0? This should be documented too. I doubt it means "no limit," does it?
The default value 0 means "no change so whatever the default value is it will be used.", I will add this information to documentation

> 0.1 would mean the database is allowed to consume 10% of the user's disk space, correct?
yes

Reply:
One more note: If we would set the ratio to 0.0 then quota be 0, so nothing would be cached/stored.

So I limit the ratio range to (0.0, 1.0>